### ISSUE TITLE
Wrong paths with multi-application modules

### ISSUE THREAD

**Issue by pedrofsn** [jeziellago/FlowNav#17]
I've two application modules, `app_a_app` and `app_b_app`. And feature modules like `app_a_payment` and `app_b_payment`. These specific modules inherit from an common scope module like `payment`.

In specific feature modules (by app), always the activity have the same name and the same EntryFlowNav like `PayersActivity`. 

So my module router do the maginc (with this cool library) calling the right path. Like `start(FEATURE_X)` inside a extension function like `openFeatureX()`.

When I'm running the app, the entrymap generated by library it's mixing paths from feature modules from other application module. 

E.g.:

[1] `br.com.app_a.ScreenCool`
[2] `br.com.app_b.ScreenCool`

I'm seeing references/paths in wrong application, causing `ActivityNotFoundException`.

**Comment by pedrofsn** [comment]
I tried removing one of modules from `flownav.main.module` but it's happening too

**Comment by jeziellago** [comment]
Hi @pedrofsn, 
I'm seeing this and soon will be fixed. 
Thanks for report.

**Comment by pedrofsn** [comment]
My guess: processor is reading all modules instead of just modules that are connected with main/application module. So the processor collect the same tag/value defined as EntryFlowNav inside getEntryMap. With this data structure, the key doesn't repeat and the value is replaced. With this, your module names can affect the experiment because some modules (with same keys) can be readed first. 

I was not able to debug the processor, so I'm just guessing here rsrs.

**Comment by jeziellago** [comment]
The kapt runs based on gradle tasks for the current build. Could you check if in your project the `:app_module:build` task run all modules?

**Comment by jeziellago** [comment]
Try runs `clean` project between application modules build.

**Comment by pedrofsn** [comment]
> The kapt runs based on gradle tasks for the current build. Could you check if in your project the `:app_module:build` task run all modules?

No, they run tasks only in right modules. If I run task for module application A, just the used modules in A were built.

**Comment by jeziellago** [comment]
> I've two application modules, `app_a_app` and `app_b_app`. And feature modules like `app_a_payment` and `app_b_payment`. These specific modules inherit from an common scope module like `payment`.
> 
> In specific feature modules (by app), always the activity have the same name and the same EntryFlowNav like `PayersActivity`.
> 
> So my module router do the maginc (with this cool library) calling the right path. Like `start(FEATURE_X)` inside a extension function like `openFeatureX()`.
> 
> When I'm running the app, the entrymap generated by library it's mixing paths from feature modules from other application module.
> 
> E.g.:
> 
> [1] `br.com.app_a.ScreenCool`
> [2] `br.com.app_b.ScreenCool`
> 
> I'm seeing references/paths in wrong application, causing `ActivityNotFoundException`.
   
The `EntryFlowNav` is equals for  `app_a_payment` and `app_b_payment`. Right. 
But, when your app module runs, only the `app_a_payment` **or** `app_b_payment` should runs too.
Could you show more details about your modules dependency graph?

**Comment by pedrofsn** [comment]
Hi @jeziellago, 

I tried to simplify because in total we've almost 70 modules and we're growing up as white label application with feature and feature_customized modules. And application modules.

But in general it's like that:

![modules_organization](https://www.dropbox.com/s/mpr6g3r5mxoz4rb/Diagrama1.png)

**Comment by pedrofsn** [comment]
In this image, "feature_a_barcode" it is an specific feature from "Application A" also "feature_b_investment" it is an specific feature from "Application B".

Now with "feature_XXXX_qrpay" we've specific customizations but they inherit from an common module called "feature_qrpay". 

Every module that can 'route'/navigate to some other module inherit :router module.

I don't showed in this diagram but we've a module called 'common'. This module have any basic classes like "BaseActivity", "BaseFragment" etc. And important libraries like material design, android x and other basic stuffied. The module common is almost inherited by each module.

**Comment by pedrofsn** [comment]
It's not exactly an problem but I found EntryFlowNav generated in one specific feature from one specific app in OTHER compiled/runned module app. 

The "other app" never will call this feature because it don't know about this feature. So I'll never get runtime error like ClassNotFound (or something like that).

**Comment by pedrofsn** [comment]
> Execution failed for task ':app_xxx_app:mergeReleaseJavaResource'.
> A failure occurred while executing com.android.build.gradle.internal.tasks.Workers$ActionFacade
> More than one file was found with OS independent path 'META-INF/router_release.kotlin_module'


I put this snippet bellow in packingOptions in each application module to solve this error. 

> exclude 'META-INF/library_release.kotlin_module'
> exclude 'META-INF/router_release.kotlin_module'

Maybe I'm putting things that doesn't matter. But I don't know what is causing trouble here, so I think I need provide maximum details. =]

**Comment by jeziellago** [comment]
@pedrofsn Thank you for sharing a lot of details about the problem. Even with some limitations, you got success with FlowNav in multi-application modules?

**Comment by pedrofsn** [comment]
Hi @jeziellago,

I moved to other branch/feature and I'm not able to work back so soon in migration to FlowNav.

I couldn't make FlowNav work correctly in a project with multi modules - with two applications module.

I drew  again some modules pursuing recursirve or wronge references. I found some parts and fixed. But generated classes are mixing activities/modules not called in gradle application module yet. 

I didn't figured out how to fix (yet).
